backend/climateconnect_api/views/user_views.py

```diff
@@ -156,6 +156,12 @@ def post(self, request):
             user_profile.last_completed_tutorial_step = request.data[
                 "last_completed_tutorial_step"
             ]
+        if "hub" in request.data:
+            hub = Hub.objects.filter(url_slug=request.data["hub"])
```

Comment:
I would recommend doing this

```python
hub = Hub.objects.filter(url_slug=requrst.data["hub"]).first()
if hub:
     user_profile.related_hubs.add(hub)
```
Reduced the number of queries you will be making to the database.

Reply:
good point. adopted changes